### PR TITLE
reproc 14.2.3 (new formula)

### PR DESCRIPTION
**Pull request by alebcay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/reproc.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  url "https://github.com/DaanDeMeyer/reproc/archive/refs/tags/v14.2.3.tar.gz"
+  sha256 "ea87db7a780a81bc27bce3e6c8e5f31b831fb8a1d06c6b5c2e1c2554d1c55e00"
+  license "MIT"
+  head "https://github.com/DaanDeMeyer/reproc.git", branch: "main"
+
+  depends_on "cmake" => :build
+
+  def install
+    args = *std_cmake_args << "-DREPROC++=ON"
+    system "cmake", "-S", ".", "-B", "build", *args, "-DBUILD_SHARED_LIBS=ON"
```
Comment (carlocab) [inline Formula/reproc.rb:13]:
```suggestion
    system "cmake", "-S", ".", "-B", "build", *args, "-DBUILD_SHARED_LIBS=ON", "-DCMAKE_INSTALL_RPATH=#{rpath}"
```

File: Formula/reproc.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    system "cmake", "--build", "build"
+    lib.install "build/reproc/lib/libreproc.a", "build/reproc++/lib/libreproc++.a"
+
+    # Rewrite variable-referenced linkage as direct linkage
+    on_macos do
+      reproc = (lib/"libreproc.dylib").readlink
+      macho = MachO.open(lib/"libreproc++.dylib")
+      macho.change_dylib("@rpath/#{reproc}", "#{lib}/#{reproc}")
+      macho.write!
+    end
```
Comment (carlocab) [inline Formula/reproc.rb:27]: I don't think you need this if you just set `CMAKE_INSTALL_RPATH` instead.

**Comment by alebcay** [comment]
The C++ test is failing to compile in Linux. It seems to be working for me in the Homebrew Docker container though, so going to have to dig a bit more and figure out what's happening.

**Comment by alebcay** [comment]
Okay, after installing my way through a number of old compilers in the Docker image, I got the build to fail. Looks like we need `g++` >= 6 to proceed.

**Review by carlocab — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Pull request closed without merging**